### PR TITLE
Header file refactoring and cleaning

### PR DESCRIPTION
**Pull request by kukkamario**
- Split Emu/Memory into more logical headers 
  - Add vm_locking.h and vm_reservation.h and move relevant functions
    and types to these headers.
  - Change include order and make vm_ptr.h, vm_var.h and vm_ref.h headers
    usable invidually and them including vm.h instead of other way around
  - Because usage of vm::ptr now requires including vm_ptr.h instead of
    vm.h updated multiple #includes
  - Added additional #includes to vm_reservation.h and vm_locking to
    where vm::reservation_* and locking related functions are used
- Add missing #includes to header files
  - Multiple header files where missing #includes to other headers that
    where used in the header. Correct header was included in correct
    order in source files which caused everything to compile.
  - Added missing #includes so header files correctly include all their
    dependencies and fixes problems with IDEs being unable to parse
    headers correctly due to missing symbols
- Removed unnecessary includes
  - Manually went through most of the source and header files and tried to find include directives to headers that were not required for compiling that file.
  - Focused most on RSX because it seemed to have most unnecessary includes and large include lists
  - Focused on removing includes that are completely unused by the file
    - Removing includes that are duplicates because of some other required header including the same header is not necessary. They don't affect compile times and don't introduce unnecessary symbols to the compile unit. Also files shouldn't rely on other headers including  headers they need.

### CODE REVIEW

File: rpcs3/Emu/RSX/Common/texture_cache.h [hunk 1]
```diff
@@ -1,7 +1,8 @@
-﻿#pragma once
+#pragma once
 
 #include "../rsx_cache.h"
 #include "../rsx_utils.h"
+#include "../RSXThread.h"
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Common/texture_cache.h]: This implies a refactor is needed. This dependency should not exist. Likely means some enum needs to be moved.

File: rpcs3/Emu/RSX/GL/GLGSRender.h [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include "GLOverlays.h"
 #include "../rsx_utils.h"
 #include "../rsx_cache.h"
+#include "../Common/texture_cache.h"
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/GLGSRender.h]: GLTextureCache.h is included above which definately includes texture_cache.h. The explicit imports of rsx_utils and rsx_cache above this line are not needed either but likely predate the shared texture cache.

File: rpcs3/Emu/RSX/GL/GLProgramBuffer.h [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #pragma once
 #include "GLVertexProgram.h"
 #include "GLFragmentProgram.h"
+#include "GLHelpers.h"
+#include "../RSXThread.h"
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/GLProgramBuffer.h]: This implies a refactor; rsx::limits should be moved to rsx_utils.h

File: rpcs3/Emu/RSX/GL/GLTextureCache.h [hunk 4]
```diff
@@ -17,8 +17,10 @@
 #include "GLOverlays.h"
 #include "../Common/TextureUtils.h"
 #include "../Common/texture_cache.h"
+#include "../Common/texture_cache_utils.h"
 #include "../../Memory/vm.h"
 #include "../rsx_utils.h"
+#include "../rsx_cache.h"
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/GLTextureCache.h]: texture_cache.h already includes rsx_utils and rsx_cache. This also goes for the existing import of rsx-utils above this one.

File: rpcs3/Emu/RSX/GL/GLTextureCache.h [hunk 5]
```diff
@@ -17,8 +17,10 @@
 #include "GLOverlays.h"
 #include "../Common/TextureUtils.h"
 #include "../Common/texture_cache.h"
+#include "../Common/texture_cache_utils.h"
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/GLTextureCache.h]: texture_cache_utils is included by texture_cache.h

File: rpcs3/Emu/RSX/Common/texture_cache.h (hunk 1 above)
Comment (kukkamario) [reply rpcs3/Emu/RSX/Common/texture_cache.h]: Discord discussion result:

`rsx::get_current_renderer()->ROP_sync_timestamp` is causing the dependency and that can't be removed without refactoring which doesn't belong to this pull-request. Marking this as resolved.

**Comment by kukkamario** [comment]
Went manually through most of the files to remove most of the unnecessary #includes

Moved `enum class texture_dimension_extended` from `RSXTexture.h` to `gcm_enums.h` because it allowed a couple headers to drop include to `RSXTexture.h`

Removed 'WIP' tag

**Comment by kd-11** [comment]
RSX/GCM changes are good, but will need review by other developers before merge as it touches well outside my purview.

**Comment by AniLeo** [comment]
Also needs conflicts to be solved (cellAudio.h) and rebase

**Comment by kukkamario** [comment]
Rebased.

Only non-trivial changes are outside RSX are Emu/Memory refactoring in the first commit.

File: rpcs3/Emu/CMakeLists.txt [hunk 6]
```diff
@@ -25,7 +25,11 @@ target_link_libraries(rpcs3_emu
 		3rdparty::pugixml
 		3rdparty::gsl)
 
-set_source_files_properties("../../Utilities/JIT.cpp" PROPERTIES COMPILE_FLAGS -fno-rtti)
+if (MSVC)
+  set_source_files_properties("../../Utilities/JIT.cpp" PROPERTIES COMPILE_FLAGS /GR-)
```
Comment (Megamouse) [inline rpcs3/Emu/CMakeLists.txt]: indentation with tab

**Comment by Zangetsu38** [comment]
Need rebase and fix few conflict

**Comment by Megamouse** [comment]
@kukkamario do you want to try again?
Sorry it wasn't merged.

**Comment by kukkamario** [comment]
I'll try to check, is this still salvageable pull request next week.

**Comment by scribam** [comment]
@kukkamario Any update on this?

**Comment by kukkamario** [comment]
Okey... I finally had time to spend full day with updating my development setup and rebasing this.

**Pull request merged**